views/qrcode.pug

```diff
@@ -9,15 +9,21 @@ html
   body
     h1 The Coding Train Live Poll
     h2 QR-Code to newest Poll
-    h3.qr-question=question
-    div#placeHolder
-    template(data-pollurl=pollURL)
+    section.content
+      section.left
+        iframe#results()
+      section.right
+        h3.qr-question=question
+        div#placeHolder
+        template(data-pollurl=pollURL)
+
 
   script.
       var typeNumber = 4;
       var errorCorrectionLevel = 'L';
       var qr = qrcode(typeNumber, errorCorrectionLevel);
       var pollURL = document.querySelector('[data-pollurl]').dataset.pollurl;
+      results.src = pollURL.replace('/vote/', '/poll/');
```

Comment:
As i understand this: You forces a change from the /vote URL to /poll URL ... The QR-Code then never will include the URL to the voting itself. The function of the QR-Code is, to direct the user the the voting page itself.

Reply:
My understanding of this line was creating a new string with '/vote/' replaced with '/poll/' for the iframe that contains the results page. The `pollURL` used to add the data to the QR code is unchanged, but I can see that the order of those lines should be changed. In fact, I would like to move the `results.src` line down to the bottom of the `script.` section.

Reply:
but as i see, you've changed the original **views/qrcode.pug** file so that will have a global effect and also changes the way the /qrcode route worked before (displaying only the qr-code the the latest voting page).

I think for what you are trying to do, you would need a separate .pug file only for your needs. Or are i'am missing something here?

Reply:
I thought you meant that the QR code will not longer bring you to the voting page. It still does bring you to the voting page unless you already voted then it brings you to the results. My motivation was to have the results show live in OBS and have the QR code up for everyone to be able to go and open the voting without having to create 2 more sources in OBS, but like I said, I was only trying to make the simplest change to propose

Reply:
That right, because the QR-Code URL still contains /vote/ (you can scan you screenshot, than you will see that it say /vote/)

` results.src = pollURL.replace('/vote/', '/poll/')`

Thats because right now that section of you code doesn't have any effect at the moment and therefore we still have the old correct behavior. As soon as you replace /vote/ through /poll/ (what you're intending to do) the behavior you're explained would not work, because it's the other way around. 

/vote/ : you can submit a vote if you already done so, you'll redirected to /poll/

Reply:
I'm not sure I understand. The QR code will continue to link to the vote page which redirects you to the results once you have voted. The purpose of `replace` on the URL string is to set the iframe source to the results page. The `replace` method does not mutate the string. Once the `pollURL` is passed to the `addData` method the QR code will have the data that contains the `'/vote/'` route

Reply:
Oh boy, because `results.src = pollURL.replace('/vote/', '/poll/')` in between the  QR-Code stuff and right before `qr.addData(pollURL);` I totally was into thinking you are going for changing the QR-Code itself. Because the pollURL variable is injected into the QR-Code in the next line. But the way you did it, not pollURL itself it mutated, only the outcome is saved to  `results.src`. So sorry for that. Totally mixed up but glad we figured it now tho :D.

Reply:
I see what happened now. Glad we figured it out :octocat: